mlbench_core/cli/cli.py

```diff
@@ -803,110 +726,85 @@ def create_gcloud(
 @create_cluster.command("kind")
 @click.argument("num_workers", type=int, metavar="num-workers")
 @click.argument("release", type=str)
-@click.argument("mlbench_helm_path", type=str)
 @click.option("--registry_name", "-r", default="kind-registry", type=str)
 @click.option("--registry_port", "-p", default="5000", type=str)
+@click.option("--host_port", "-h", default="5000", type=str)
 @click.option("--num-cpus", "-c", default=1, type=int)
 @click.option("--num-gpus", "-g", default=0, type=int)
+@click.option("--custom-value", "-v", multiple=True)
 def create_kind(
     num_workers,
     release,
-    mlbench_helm_path,
     registry_name,
     registry_port,
+    host_port,
     num_cpus,
     num_gpus,
+    custom_value,
 ):
-    # check if local registry exists
-    p = subprocess.Popen(
-        ["docker", "inspect", "-f", "'{{.State.Running}}'", registry_name],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-    )
+    name = "{}-{}".format(release, num_workers)
 
-    output, error = p.communicate()
-    if p.returncode != 0:
-        raise click.UsageError(error.decode())
+    docker_client = docker.from_env()
+
+    # check if local registry exists
+    existing_containers = [
+        container.name for container in docker_client.containers.list()
+    ]
+    registry_exists = registry_name in existing_containers
 
-    if output.decode().strip().replace("'", "") != "true":
+    if not registry_exists:
         # create local registry
         click.echo("Creating registry {}".format((registry_name)))
+        docker_client.containers.run(
+            image="registry:2",
+            name=registry_name,
+            restart_policy={"Name": "always"},
+            ports={registry_port: host_port},
+            detach=True,
+        )
+        while docker_client.containers.get(registry_name).status != "running":
```

Comment:
If there is something wrong and the container immediately terminates, the status could be `exited` and this loop would hang indefinitely.

Also it's good practice to add a sleep of 0.5-1 second here so the code is not busy waiting and using up unnecessary CPU